corehq/apps/reports/generic.py

```diff
@@ -223,20 +227,24 @@ class FakeHttpRequest(object):
             datespan = None
             can_access_all_locations = None
 
+        if 'startdate' and 'enddate' in state['request_params']:
+            date_holder = state['request_params']
+        elif 'startdate' and 'enddate' in state['request']:
+            date_holder = state['request']
```

Comment:
yep! sometimes the start and enddates are in the request and sometimes in the request_params, depending on the report. This could potentially be cleaned up in the future but this is the simpler solution for now.

Reply:
I think I'm confused here. The docstring says that this function will only receive parameters from `get_json_report_parameters`, right? `get_json_report_parameters` does not treat the datespan as optional -- `.pop(property)` raises an exception if the property does not exist. So `report_state['request_params']` will always contain `start_date` and `end_date`, because `get_json_report_parameters` will always set them.

Reply:
Ahh my bad, that comment isn't strictly true and definitely should've been updated. `export_all_rows_task` which uses the `set_report_parameters` is called from multiple places and the request parameters where the dates aren't in the `request_params` come from when they're called from a task in the saved_reports app, which is what I meant by it could use some cleaning up.

Thinking back on it, I may have pulled the trigger too early on this PR - would it be better to have another PR to clarify that in the comments and/or clean it up so the dates are always in the request_params? Or revert and fix it here?

Reply:
for this specifically, I think a separate PR is fine